Week3/morejava.js

```diff
@@ -0,0 +1,181 @@
+    if (code === 1) {
+        console.log('a ' + color + ' car');
+    }
+    if (code === 2) {
+        console.log('a ' + color + ' motorbike');
+    }
+}
+vehicleType("blue", 2)
+vehicleType("red", 1)
+
+
+//Can you write the following without the if statement,
+//but with just as a single line with console.log(...);?
+
+if (3 === 3) {
+    console.log("yes");
+} else {
+    console.log("no");
+}
+
+console.log((3 === 3) ? 'yes' : 'no');
+
+//Create a function called vehicle, like before, 
+//but takes another parameter called age, so that vehicle("blue", 1, 5) prints 'a blue used car'
+
+function vehicle(color, code, years) {
+    if (code === 1 && years < 2) {
+        console.log('a ' + color + ' new car')
+
+
+    }
+    if (code == 1 && years >= 2) {
+        console.log('a ' + color + ' used car')
+
+
+    }
+    if (code == 2 && years < 2) {
+        console.log('a ' + color + ' new motorbike')
+
+
+    }
+    if (code == 2 && years >= 2) {
+        console.log('a ' + color + ' used motorbike')
+
+
+    }
+
+}
+vehicle("blue", 1, 5);
+
+//Make a list of vehicles, you can add "motorbike", "caravan", "bike", or more.
+
+listOfVehicles = ["motorbike", "caravan", "bike", "car"];
+//How do you get the third element from that list?
+listOfVehicles[2];
+console.log(listOfVehicles[2]);
+
+//Change the function vehicle to use the list of question 7.
+//So that vehicle("green", 3, 1) prints "a green new bike".
+function newVehicle(color, code, years) {
+    if (years < 2) {
+        console.log('a ' + color + ' new ' + listOfVehicles[code - 1]);
+    }
+    if (years >= 2) {
+        console.log('a ' + color + ' used ' + listOfVehicles[code - 1]);
+    }
+  
+}
+newVehicle("green", 3, 1);//a green new bike
+newVehicle("black", 4, 8);//a black used car
+
+//Use the list of vehicles to write an advertisement.
+//So that it prints something like: "Amazing Joe's Garage, we service cars, motorbikes, caravans and bikes.". (Hint: use a for loop.)
+function advertisement(listOfVehicles) {
+    let adv = "Amazing Joe's Garage, we service "
+
+    for (let i = 0; i < listOfVehicles.length; i++) {
+        adv += (listOfVehicles[i] + "s, ")
+    }
+    console.log(adv + listOfVehicles[listOfVehicles.length - 2] + "s and " + listOfVehicles[listOfVehicles.length - 1] + "s.")
+}
+advertisement(listOfVehicles);
+
+//What if you add one more vehicle to the list, can you have that added to the advertisement without changing the code for question 10?
+listOfVehicles.push('train');
+console.log(listOfVehicles);
+
+//Create an empty object.
+let obj = {};
+
+//Create an object that contains the teachers that you have had so far for the different modules.
+const teachersInfo = {
+    names: ["Philip", "Unmesh", "Yash"]
+};
+
+
+console.log(teachersInfo);
+
+//Add a property to the object you just created that contains the languages that they have taught you.
+teachersInfo.languages = ['HTML/CSS', 'CLI', 'JavaScript1'];
+console.log(teachersInfo);
```

Comment:
I find a slight problem here. Your **teachersInfo** object looks like this:
```
const teachersInfo = {
  names: ["Philip", "Unmesh", "Yash"],
  languages: ['HTML/CSS', 'CLI', 'JavaScript1']
};
```

1. You are using both double quotes(") and single quotes(') for creating strings. It's consistent if you pick one and use that everywhere. 🙂 
2. It would be hard to identify a person not knowing HYF to identify which language was taught by which teacher? For example, did Phillip teach JS1 or Unmesh HTML/CSS? 😆 

It's important that we create meaningful data structures to represent information we want to create. Wouldn't the following be a better approach to this?

```
const teachersInfo = {
  Phillip: 'HTML/CSS',
  Unmesh: 'CLI',
  Yash: 'JavaScript1',
};
```

Reply:
Done! 👍